Yosemite/Yosemite/Model/Payments/CardPresentPaymentsConfiguration.swift

```diff
@@ -1,64 +1,73 @@
+
     public let paymentMethods: [WCPayPaymentMethodType]
     public let currencies: [String]
     public let paymentGateways: [String]
     public let supportedReaders: [CardReaderType]
-    public let purchaseCardReaderUrl: URL
-
 
-    init(paymentMethods: [WCPayPaymentMethodType],
+    init(countryCode: String,
+         paymentMethods: [WCPayPaymentMethodType],
          currencies: [String],
          paymentGateways: [String],
-         supportedReaders: [CardReaderType],
-         purchaseCardReaderUrl: URL) {
+         supportedReaders: [CardReaderType]) {
+        self.countryCode = countryCode
         self.paymentMethods = paymentMethods
         self.currencies = currencies
         self.paymentGateways = paymentGateways
         self.supportedReaders = supportedReaders
-        self.purchaseCardReaderUrl = purchaseCardReaderUrl
     }
 
     public init(country: String, canadaEnabled: Bool) {
         switch country {
         case "US":
-            //TODO: update to use Self.purchaseCardReaderUrl(for: country) when pages/redirects are added to the website pdfdoF-su-p2
             self.init(
+                countryCode: country,
                 paymentMethods: [.cardPresent],
                 currencies: ["USD"],
                 paymentGateways: [WCPayAccount.gatewayID, StripeAccount.gatewayID],
-                supportedReaders: [.chipper, .stripeM2],
-                purchaseCardReaderUrl: Constants.purchaseM2ReaderUrl
+                supportedReaders: [.chipper, .stripeM2]
             )
         case "CA" where canadaEnabled == true:
             self.init(
+                countryCode: country,
                 paymentMethods: [.cardPresent, .interacPresent],
                 currencies: ["CAD"],
                 paymentGateways: [WCPayAccount.gatewayID],
-                supportedReaders: [.wisepad3],
-                purchaseCardReaderUrl: Self.purchaseCardReaderUrl(for: country)
+                supportedReaders: [.wisepad3]
             )
         default:
-            self.init(paymentMethods: [],
-                      currencies: [],
-                      paymentGateways: [],
-                      supportedReaders: [],
-                      purchaseCardReaderUrl: Constants.fallbackInPersonPaymentsUrl)
+            self.init(
+                countryCode: country,
+                paymentMethods: [],
+                currencies: [],
+                paymentGateways: [],
+                supportedReaders: []
+            )
         }
     }
 
     public var isSupportedCountry: Bool {
         paymentMethods.isEmpty == false && currencies.isEmpty == false && paymentGateways.isEmpty == false && supportedReaders.isEmpty == false
     }
 
-    private static func purchaseCardReaderUrl(for countryCode: String) -> URL {
-        URL(string: Constants.purchaseReaderForCountryUrlBase + countryCode) ?? Constants.fallbackInPersonPaymentsUrl
+    /// Given a two character country code and the active plugin, returns a URL
```

Comment:
@koke - let me know if you have any objections to this change

Reply:
I think the approach is fine (plugin-dependent function instead of property), but @joshheald kept the M2 URL in https://github.com/woocommerce/woocommerce-ios/pull/6216#issuecomment-1044217612 because the country-specific URLs are not live yet.

Also, now that I think of it, whenever we add a new country, the new code would attempt to use the country-based URL instead of the fallback one, and I'm not sure if that's what we want. If we make sure that the backend can always redirect all the country codes to a relevant page then it's fine. In any case, it's not something that we'd release until the URL was set up, so it doesn't seem that important.